src/lib/services/precommit.js

```diff
@@ -77,9 +81,22 @@ class Precommit {
     }
   }
 
+  isFileToBeCommitted (filePath) {
+    try {
+      const output = execSync('git diff --cached --name-only').toString()
+      const files = output.split('\n')
+      return files.includes(filePath)
+    } catch (error) {
+      console.error('Error checking if file is to be committed:', error)
+      return false
+    }
+  }
+
```

Comment:
I'm not sure if it should be taken for granted that the user is using git...
Maybe the Precommit class should be configured to support several version control systems and default to git?

Reply:
I agree but I would need suggestions so about how this is supposed to be fixed :( I could suggest to check if git is used and if not fallback to previous check

Reply:
I added a possible fix for this so at worse this will keep working like it is now

Reply:
that fallback is good. safe to safe that prob 90% of those using it currently are using git. likely even higher.